src/mono/mono/mini/interp/transform.c

```diff
@@ -2723,6 +2724,12 @@ interp_inline_method (TransformData *td, MonoMethod *target_method, MonoMethodHe
 			tmp_bb->il_offset = prev_ip - prev_il_code;
 			tmp_bb = tmp_bb->next_bb;
 		}
+		// After inlining, new blocks have been added as continuation to the original 'prev' block.
+		// Once the inlining is complete, we need to mark the original block as emitted as well.
+		// NOTE: There is also a potential problem with:
+		// td->offset_to_bb[td->cdd->il_offset] which will not point to the newly added block (containing inlined call),
```

Comment:
I don't understand why this is a problem? we shouldn't need to look at the old call instruction again, I think.
should we zero it out in `offset_to_bb` ?

Reply:
The reason for the comment comes from the fact that the new code gen approach combines two different ways of advancing a basic block:
1. The original approach ie `first pass` approach - where `td->offset_to_bb` maps offsets to basic blocks and uses this information to advance, by reading the map with current `in_offset`

2. The new approach ie `next pass` approach - where `td->cbb->next` is used for advancing (when skipping already emitted blocks)

After inlining, Inlined blocks are not mapped to the `td->offset_to_bb`. They exist as next blocks of the caller block. Additionally, once the inlining is completed the `td->cbb` is actually the last inlined block.  

All this seemed as something that could cause issues with the new approach. I will rethink if that comment is still valid or not.